php/php.blade/src/org/netbeans/modules/php/blade/syntax/antlr4/v10/BladeAntlrLexer.java

```diff
@@ -0,0 +1,2607 @@
+		switch (predIndex) {
+		case 5:
+			return this._input.LA(1) != '"';
+		}
+		return true;
+	}
+	private boolean HTML_L_COMPONENT_sempred(RuleContext _localctx, int predIndex) {
+		switch (predIndex) {
+		case 6:
+			return this._input.LA(1) == '>';
+		}
+		return true;
+	}
+	private boolean WS_EXPR_ESCAPE_sempred(RuleContext _localctx, int predIndex) {
+		switch (predIndex) {
+		case 7:
+			return this._input.LA(1) == '@';
+		}
+		return true;
+	}
+	private boolean L_OHTER_ESCAPE_sempred(RuleContext _localctx, int predIndex) {
+		switch (predIndex) {
+		case 8:
+			return this._input.LA(1) == '@';
+		}
+		return true;
+	}
+	private boolean OPEN_EXPR_PAREN_sempred(RuleContext _localctx, int predIndex) {
+		switch (predIndex) {
+		case 9:
+			return this.roundParenBalance == 0;
+		}
+		return true;
+	}
+	private boolean CLOSE_EXPR_PAREN_sempred(RuleContext _localctx, int predIndex) {
+		switch (predIndex) {
+		case 10:
+			return this.roundParenBalance == 1;
+		}
+		return true;
+	}
+	private boolean LPAREN_sempred(RuleContext _localctx, int predIndex) {
+		switch (predIndex) {
+		case 11:
+			return this.roundParenBalance > 0;
+		}
+		return true;
+	}
+	private boolean RPAREN_sempred(RuleContext _localctx, int predIndex) {
+		switch (predIndex) {
+		case 12:
+			return this.roundParenBalance > 0;
+		}
+		return true;
+	}
+	private boolean EXIT_RPAREN_sempred(RuleContext _localctx, int predIndex) {
+		switch (predIndex) {
+		case 13:
+			return this.roundParenBalance == 0;
+		}
+		return true;
+	}
+	private boolean PHP_D_NEW_sempred(RuleContext _localctx, int predIndex) {
+		switch (predIndex) {
+		case 14:
+			return this._input.LA(1) == ' ';
+		}
+		return true;
+	}
+	private boolean VERBATIM_HTML_sempred(RuleContext _localctx, int predIndex) {
+		switch (predIndex) {
+		case 15:
+			return 
+		        this._input.LA(1) == '@' &&
+		        this._input.LA(2) == 'e' &&
+		        this._input.LA(3) == 'n' &&
+		        this._input.LA(4) == 'd' &&
+		        this._input.LA(5) == 'v' &&
+		        this._input.LA(6) == 'e' &&
+		        this._input.LA(7) == 'r'
+		      ;
+		}
+		return true;
+	}
+	private boolean BLADE_COMMENT_PEEK_sempred(RuleContext _localctx, int predIndex) {
+		switch (predIndex) {
+		case 16:
+			return 
+		        this._input.LA(1) == '-' &&
+		        this._input.LA(2) == '-' &&
+		        this._input.LA(3) == '}' &&
+		        this._input.LA(4) == '}'
+		      ;
+		}
+		return true;
+	}
+
+	private static final String _serializedATNSegment0 =
+		"\u0004\u0000\u00ab\u0b4c\u0006\uffff\uffff\u0006\uffff\uffff\u0006\uffff"+
+		"\uffff\u0006\uffff\uffff\u0006\uffff\uffff\u0006\uffff\uffff\u0006\uffff"+
+		"\uffff\u0006\uffff\uffff\u0006\uffff\uffff\u0006\uffff\uffff\u0006\uffff"+
+		"\uffff\u0006\uffff\uffff\u0006\uffff\uffff\u0006\uffff\uffff\u0002\u0000"+
+		"\u0007\u0000\u0002\u0001\u0007\u0001\u0002\u0002\u0007\u0002\u0002\u0003"+
+		"\u0007\u0003\u0002\u0004\u0007\u0004\u0002\u0005\u0007\u0005\u0002\u0006"+
```

Comment:
regarding generated code. I believe we ended up agreeing to try to generate antlr outputs during build but don't commit it to the repo. The reason is because generated antlr code is not very "stable", it produces huge diffs when the version updates and those aren't reviewable anyway. (e.g the marked section)

The second reason would be antlr updates, this makes sure the code fits to the version if someone happens to update the antlr lib at some point.

example: https://github.com/apache/netbeans/pull/7189, https://github.com/apache/netbeans/pull/7186 etc

cc @neilcsmith-net @lkishalmi

Reply:
Ok, I will see how I can add it to .gitignore and have them build before the unit tests.
@junichi11 is it ok for you if I remove the generated antlr parser and lexer files from git ?

Reply:
Hi @mbien 
I've removed the generated antlr *.java files from the repo and set the build task for antlr lexer & parser as a extension of `extensionOf="-pre-compile"`

```
<target name="generate-antlr-parser" description="Regenerate the ANTLRv4 parser" depends="init" extensionOf="-pre-compile">
```

Seems to work fine.

Reply:
@haidubogdan awesome!